### PR TITLE
Added default values to game config form

### PR DESCRIPTION
**Pull request by yousuf031991**
- Default values added wrt Android app
- Added front end validation to remove negative values
- Changed mongoose schema for better validation
- Changed backend validation, removed usage of mongoose-validator

### CODE REVIEW

File: MainApp/lib/app/routes/api.js [hunk 1]
```diff
@@ -52,30 +52,35 @@ export default function (router) {
...
+
+    //http://localhost:8080/api/gameinfo
+    router.post('/gameinfo', function (req, res) {
+        let gameinfo = new Game();
+        gameinfo.gameConfigId = req.body.gameConfigId;
+        gameinfo.trialInfoId = req.body.trialInfoId;
+        gameinfo.userStatsId = req.body.userStatsId;
+        gameinfo.username = req.body.username;
+
+        if (gameinfo.gameConfigId == null || gameinfo.gameConfigId == '' || gameinfo.trialInfoId == null || gameinfo.trialInfoId == '' || gameinfo.userStatsId == null || gameinfo.userStatsId == '') {
```
Comment (super9user) [inline MainApp/lib/app/routes/api.js:74]: Can't we have the same model level validation as used in gameConfig?

**Comment by super9user** [comment]
+1 (with a minor non-blocking comment)

File: MainApp/lib/app/routes/api.js (hunk 1 above)
Comment (yousuf031991) [reply MainApp/lib/app/routes/api.js:74]: Yeah we should change all those. Apparently the mongoose-validator library is kind of useless. 
Here's a [good tutorial](http://thecodebarbarian.com/2015/02/20/better-angularjs-form-validation-with-mongoose) I found for validation.
I think Tharun is handling this schema.

Comment (super9user) [reply MainApp/lib/app/routes/api.js:74]: Ok , can we add that as an issue in github. So that we don't forget to do it later

**Pull request merged**